Jetpack Partner Portal: implement URL support to show more info modal about the license in the issue license page

yashwin opened the pull request:
Related to 1202619025189113-as-1204942360069653

## Proposed Changes

This PR implements URL support to show more info modal about the license in the /partner-portal/issue-license page. You can now append the URL(/partner-portal/issue-license) with `show_license_modal={product_slug}`, which will open the product info modal.

#### Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Run `git checkout add/url-support-open-license-info-popup` and `yarn start-jetpack-cloud` or open the Jetpack Cloud live link. 
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard.
3. Visit http://jetpack.cloud.localhost:3001/partner-portal/issue-license?show_license_modal=jetpack-ai and verify that the product info modal for AI Assistant opens up. Verify the same for other products(jetpack-scan) and bundles - http://jetpack.cloud.localhost:3001/partner-portal/issue-license?show_license_modal=jetpack-complete
4. Click the `More about {product}` button and verify that the URL is appended with the query param.

<img width="646" alt="Screenshot 2023-06-30 at 7 14 07 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/f7b9e5c6-ea28-46a3-afcf-ee2d27bff922">

<img width="1257" alt="Screenshot 2023-06-30 at 7 14 11 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/ae387911-855a-4d47-b26f-2cd34d5d2d69">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

yashwin committed:
Implement URL support to show more info modal about the license

jkguidaven commented on client/jetpack-cloud/sections/partner-portal/issue-multiple-licenses-form/index.tsx:
```diff
@@ -71,6 +71,8 @@ export default function IssueMultipleLicensesForm( {
 		[]
 	);
 
+	const licenseModalToBeShown = getQueryArg( window.location.href, 'show_license_modal' );
```
In the pricing page, we used a hash fragment (e.g. https://cloud.jetpack.com/pricing#jetpack_starter_yearly)

 to determine which modal to display, as this usually meant we were trying to reach certain elements on the page.

What do you think of using hash fragment?

yashwin replied:
I feel like it is better to use query parameters in this case. `hash` is mostly used when there is an identifier(id) set to an element. I also did some research, and it is recommended to use query parameters for the modal. So, I'm going to skip this.

jkguidaven commented on client/jetpack-cloud/sections/partner-portal/license-product-card/index.tsx:
```diff
@@ -32,9 +33,10 @@ export default function LicenseProductCard( props: Props ) {
 	} = props;
 	const productTitle = getProductTitle( product.name );
-	const [ showLightbox, setShowLightbox ] = useState( false );
+	const [ showLightbox, setShowLightbox ] = useState( showLicenseInfoModal );
```
Are there reasons why we can't query the URL directly here to set the default state?

yashwin replied:
Fixed here - https://github.com/Automattic/wp-calypso/pull/78850/commits/9a44723c4e9364c9294c3f63689b1e1d34bdb738

jkguidaven commented on client/jetpack-cloud/sections/partner-portal/license-bundle-card/index.tsx:
```diff
@@ -16,12 +16,13 @@ interface Props {
 	const { tabIndex, product, onSelectProduct } = props;
 	const productTitle = getProductTitle( product.name );
-	const [ showLightbox, setShowLightbox ] = useState( false );
+	const [ showLightbox, setShowLightbox ] = useState( props.showLicenseInfoModal );
```
When we open a modal, we should also update the URL so the user can easily share the correct modal link. wdyt?

yashwin replied:
That is a very good point, @jkguidaven. I have made the changes here - https://github.com/Automattic/wp-calypso/pull/78850/commits/9a44723c4e9364c9294c3f63689b1e1d34bdb738

jkguidaven reviewed (commented):
Thanks for working on this @yashwin. It tested well as describe in the test instructions. 👍 

I left some comments.

yashwin committed:
Address PR comments

yashwin commented:
> Thanks for working on this @yashwin. It tested well as describe in the test instructions. 👍

Thanks for your thorough review, @jkguidaven!

> I left some comments.

I have addressed all your comments 👍

jkguidaven reviewed (approved):
Thanks for addressing all my comments. LGTM 👍 nice work
